### PR TITLE
hudson.plugins.ec2.win package spring clean-up

### PR DESCRIPTION
**Pull request by karliszigurs**
This is a more extensive follow-up to https://github.com/jenkinsci/ec2-plugin/pull/263 which, in addition to fixing the Piped streams use, also includes lots of small code cleanups of whole `hudson.plugins.ec2.win` package.

* Original issue (included in here) described in https://github.com/jenkinsci/ec2-plugin/pull/263
* Code cleanups, dead code cleanup, general formatting cleanups, etc...
* Care and judgement was exercised to make sure that no functional changes are introduced

This represents custom build of the plugin we rely on now internally.

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java [hunk 1]
```diff
@@ -65,20 +62,17 @@ protected void launch(EC2Computer computer, TaskListener listener, Instance inst
...
                 public void onClosed(Channel channel, IOException cause) {
                     process.destroy();
-                    connection.close();
                 }
             });
         } catch (Throwable ioe) {
             logger.println("Ouch:");
             ioe.printStackTrace(logger);
-        } finally {
-            connection.close();
```
Comment (Jimilian) [inline src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java]: why it's not needed anymore? (I'm not a maintainer, just curious :) )

**Comment by steven-foster** [comment]
Does this bring any performance improvements to the remoting channel over winrm? Mentions of FastPipedStreams are getting my hopes up... 
spent a lot of time to get the connection to agents working and found the speed unusably slow, but it would be a pity to go back to ssh

**Comment by steven-foster** [comment]
Seems not, tried it out and it still takes about 10 minutes to actually start checking out a pipeline job

File: src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java [hunk 2]
```diff
@@ -8,34 +10,29 @@
...
 
-import com.amazonaws.AmazonClientException;
-import com.amazonaws.services.ec2.model.Instance;
-
 public class EC2WindowsLauncher extends EC2ComputerLauncher {
+
     private static final String SLAVE_JAR = "slave.jar";
-    
-    final long sleepBetweenAttemps = TimeUnit.SECONDS.toMillis(10);
+    private static final long DEFAULT_RETRY_DELAY_SECONDS = 10;
```
Comment (francisu) [inline src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java]: I like the constant, but let's also keep track of the ms (as a constant or field, does not matter) here. See comment below.

File: src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java [hunk 3]
```diff
@@ -133,9 +128,10 @@ private WinConnection connectToWinRM(EC2Computer computer, PrintStream logger) t
                 WinConnection connection = new WinConnection(ip, computer.getNode().remoteAdmin, computer.getNode().getAdminPassword().getPlainText());
 
                 connection.setUseHTTPS(computer.getNode().isUseHTTPS());
+
                 if (!connection.ping()) {
-                    logger.println("Waiting for WinRM to come up. Sleeping 10s.");
-                    Thread.sleep(sleepBetweenAttemps);
+                    logger.println("Waiting for WinRM to come up. Sleeping " + DEFAULT_RETRY_DELAY_SECONDS + "s.");
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(DEFAULT_RETRY_DELAY_SECONDS));
                     continue;
```
Comment (francisu) [inline src/main/java/hudson/plugins/ec2/win/EC2WindowsLauncher.java]: Please use a single field (like before) for the ms value, I don't like the long line of duplication of the conversion code.

**Comment by thoulen** [comment]
Can you rebase ?

**Comment by karliszigurs** [comment]
obsolete

**Pull request closed without merging**